docs/users/extend/custom-commands.md

```diff
@@ -104,13 +99,92 @@ Useful variables for container scripts are:
 
 ### Annotations supported
 
-The custom commands support various annotations in the header which are used to provide additional information about the command to the user:
+The custom commands support various annotations in the header which are used to provide additional information about the command to the user.
+
+#### Description
+
+`Description` is used for the listing of available commands and for the help message of the custom command.
+
+Usage: `## Description: <command-description>`
+Example: `## Description: my great custom command`
+
+#### Usage
+
+`Usage` is used for the help message to provide an idea to the user how to use this command.
+
+Usage: `## Usage: <command-usage>`
+Example: `## Usage: commandname [flags] [args]`
+
+#### Example
+
+`Example` is used for the help message to provide some usage examples to the user. Use `\n` to force a line break.
+
+Usage: `## Example: <command-example>`
+Example: `## Example: commandname\ncommandname -h`
+
+#### Flags
+
+`Flags` is used for the help message. All defined flags here are listed with their shorthand if available. It has to be encoded according the following definition:
```

Comment:
This should be a bit stronger. I know you're planning another PR or two, so let's not delay this one, but when you next work on commands, let's make this say something like "THIS DOES NOT DEFINE THE FLAG HANDLING, ONLY WHAT COMES OUT IN THE HELP"

Reply:
Yes, agree...